aioesphomeapi/model.py

```diff
@@ -207,26 +207,46 @@ class FanState(EntityState):
 
 
 # ==================== LIGHT ====================
+class LightColorMode(APIIntEnum):
+    UNKNOWN = 0
+    WHITE = 1
+    COLOR_TEMPERATURE = 3
+    COLD_WARM_WHITE = 4
+    RGB = 8
+    RGB_WHITE = 9
+    RGB_COLOR_TEMPERATURE = 11
+    RGB_COLD_WARM_WHITE = 12
+
+
 @dataclass(frozen=True)
 class LightInfo(EntityInfo):
     supports_brightness: bool = False
-    supports_rgb: bool = False
-    supports_white_value: bool = False
-    supports_color_temperature: bool = False
+    supported_color_modes: List[LightColorMode] = converter_field(
+        default_factory=list, converter=list
+    )
     min_mireds: float = 0.0
     max_mireds: float = 0.0
     effects: List[str] = converter_field(default_factory=list, converter=list)
 
+    # deprecated, do not use
```

Comment:
Let's call these `legacy_...`.

Then let's add a `supported_color_modes_compat` method that assembles the `color_mode` values given the peer API version (like with climate presets). Same for the call in `client.py`.

Advantage then is that we have all the compat logic in aioesphomeapi, and not in HA.

Reply:
Ok scratch that about `client.py` - the client class doesn't have enough info to transform those calls. But for `LightInfo` we can do it:

```python3
    legacy_supports_rgb: bool = False
    legacy_supports_white_value: bool = False
    legacy_supports_color_temperature: bool = False

    def supported_color_modes_compat(self, api_version: APIVersion) -> List[LightColorMode]:
        if api_version < APIVersion(1, 6):
            key = (
                self.legacy_supports_rgb, 
                self.legacy_supports_white_value, 
                self.legacy_supports_color_temperature
            )
            # map legacy flags to color modes, 
            # key: (rgb, white, color_temp)
            modes_map = {
                (False, False, False): [],
                (False, False, True): [LightColorMode.COLOR_TEMPERATURE],
                (False, True, False): [],  # invalid
                (False, True, True): [],  # invalid
                (True, False, False): [LightColorMode.RGB],
                (True, False, True): [],  # invalid
                (True, True, False): [LightColorMode.RGB_WHITE],
                (True, True, True): [LightColorMode.RGB_COLOR_TEMPERATURE]
            }
            return modes_map[key]

        return self.supported_color_modes
```

Reply:
> Let's call these `legacy_...`.

Just to be sure, we don't care about backward compatibility here?

Reply:
Not really, the package is only really used by HA afaik (and some smaller things, but none that would touch color modes I think). We can just create a major version bump.